core/src/main/scala/org/bitcoins/core/psbt/PSBT.scala

```diff
@@ -0,0 +1,336 @@
+package org.bitcoins.core.psbt
+
+import org.bitcoins.core.crypto.{ECDigitalSignature, ECPublicKey, ExtPublicKey}
+import org.bitcoins.core.hd.BIP32Path
+import org.bitcoins.core.number.UInt32
+import org.bitcoins.core.protocol.CompactSizeUInt
+import org.bitcoins.core.protocol.script.{
+  EmptyScriptSignature,
+  ScriptPubKey,
+  ScriptSignature,
+  ScriptWitness,
+  WitnessScriptPubKey
+}
+import org.bitcoins.core.protocol.transaction.{
+  BaseTransaction,
+  EmptyWitness,
+  Transaction,
+  WitnessTransaction
+}
+import org.bitcoins.core.script.crypto.HashType
+import scodec.bits._
+
+object PSBT {
+  final def magicBytes = hex"70736274ff"
+
+  def fromBytes(byteVector: ByteVector): PSBT = {
+    require(byteVector.take(5) == hex"70736274ff",
+            "A PSBT must start with the magic bytes 0x70736274ff")
```

Comment:
again, with these invariants you will thank yourself later if you include the input in the error message so 

```scala
    require(byteVector.take(5) == hex"70736274ff",
            s"A PSBT must start with the magic bytes 0x70736274ff, got=${byteVector}")
```